docs/topics/multiplatform/multiplatform-advanced-project-structure.md

```diff
@@ -168,6 +168,32 @@ There are three important concepts in dependency resolution:
 
    ![Error on JVM-specific API in common code](dependency-resolution-error.png){width=700}
 
+### Aligning versions of common dependencies across source sets
+
+Common dependencies need to be aligned across source sets to make sure that common code is always compiled against
+the version of a library that satisfies all source sets.
+
+In the example above, imagine that you want to add the `androidx.navigation:navigation-compose:2.7.7` dependency to your
+`androidMain` source set. Your project explicitly declares the `kotlinx-coroutines-core:1.7.3` dependency for the `commonMain`
+source set, but the Compose Navigation library with the version 2.7.7 requires Kotlin coroutines 1.8.0 or newer.
+
+Resolving this, Gradle applies `kotlinx-coroutines-core:1.8.0` to the `commonMain` source set. But to make the common code
```

Comment:
> Gradle applies ... to the `commonMain` source set.

It is unclear why Gradle does that? i mean reader can struggle to infer that information from the sentence above.
So maybe it worth adding some explanation. Something like that:

Since commonMain source set is compiled together with androidMain, it means that commonMain should see the same version of `kotlinx-coroutines-core` as androidMain.  Despite the fact that its direct dependency was just `kotlinx-coroutines-core:1.7.3`.